doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,597 @@
+
+.. math::
+
+    o = \left\lfloor \frac{i + 2p - k}{s} \right\rfloor + 1.
+
+Transposed convolution
+----------------------
+
+If a transposed convolution is specified by
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p`,
+
+then the output size is given by
+
+.. math::
+
+    o = s (i - 1) + a + k - 2p, \quad a \in \{0, \ldots, s - 1\}
+
+where :math:`a` is a user-specified quantity used to distinguish between the
+:math:`s` different possible output sizes.
+
+Refresher: discrete convolution
+===============================
+
+The bread and butter of neural networks is affine transformations: a vector is
+received as input and is multiplied with a matrix to produce an output (to which
+a bias vector is usually added before passing the result through a
+nonlinearity). This is applicable to any type of input, be it an image, a sound
+clip or an unordered collection of features: their representation can always be
+flattened into a vector before the transformation.
+
+In the case of images and sound clips, there is some ordering (spatial or
+temporal) that is thrown away when they are flattened. This information may
+prove very handy in solving some tasks, like computer vision and speech
+recognition.
+
+These sorts of inputs share important properties:
+
+* They are stored as N-dimensional arrays (called *tensors* in Theano
+  terminology).
+* They feature one or more axes for which ordering matters (e.g. width and
+  height axes for an image, time axis for a sound clip).
+* One axis, the channel axis, is used to access different views of the data
+  (e.g. the red, green and blue channels of a color image, or the left and
+  right channels of a stereo audio track).
+
+A discrete convolution is a linear transformation that has the following
+properties:
+
+* It is sparse, i.e. only a few input units contribute to a given output unit.
+* Parameters are shared, i.e. the same transformation is applied to multiple
+  locations in the input.
+
+Here is an example of a discrete convolution:
+
+.. figure:: conv_arithmetic_figures/numerical_no_padding_no_strides.gif
+
+The blue grid is called the input *feature map*. We slide a *kernel* (dark blue)
+with value
+
+.. math::
+
+    \begin{pmatrix}
+    0 & 1 & 2 \\
+    2 & 2 & 0 \\
+    0 & 1 & 2
+    \end{pmatrix}
+
+across the input feature map. At each location, we compute the product between
+the kernel elements and the input elements they overlap and sum the results up
+to obtain the corresponding output *feature map* value.
+
+.. note::
+
+    What we referred to earlier as *channels* is a name for the *feature maps*
+    of observed data.
+
+If there are multiple input feature maps, each input feature map is convolved
+with a distinct kernel, and the resulting feature maps are summed up elementwise
+to produce the output feature map. The procedure can be repeated using different
+kernels to form as many output feature maps as desired.
```

Comment:
Finally (then I promise I'll move to the next paragraph) this would make much clearer the last sentence about using different kernels. As it currently is, we say that we use one distinct kernel for each channel...and then that we can also use different kernels to get more feature maps. This confusion wouldn't arise if we talk about *one* 3D kernel.